mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2019,19 +2019,99 @@ pub async fn send_maker_refunds_payment<T: UtxoCommonOps + SwapOps>(
     refund_htlc_payment(coin, args).await.map(|tx| tx.into())
 }
 
-/// Extracts pubkey from script sig
-fn pubkey_from_script_sig(script: &Script) -> Result<H264, String> {
+/// Extracts the signature from a scriptSig at instruction 0.
+///
+/// Usable for P2PK and P2PKH scripts.
+fn extract_signature(script: &Script) -> Result<Vec<u8>, String> {
     match script.get_instruction(0) {
         Some(Ok(instruction)) => match instruction.opcode {
             Opcode::OP_PUSHBYTES_70 | Opcode::OP_PUSHBYTES_71 | Opcode::OP_PUSHBYTES_72 => match instruction.data {
-                Some(bytes) => try_s!(SecpSignature::from_der(&bytes[..bytes.len() - 1])),
-                None => return ERR!("No data at instruction 0 of script {:?}", script),
+                Some(bytes) => Ok(bytes.to_vec()),
+                None => ERR!("No data at instruction 0 of script {:?}", script),
             },
-            _ => return ERR!("Unexpected opcode {:?}", instruction.opcode),
+            _ => ERR!("Unexpected opcode {:?}", instruction.opcode),
         },
-        Some(Err(e)) => return ERR!("Error {} on getting instruction 0 of script {:?}", e, script),
-        None => return ERR!("None instruction 0 of script {:?}", script),
+        Some(Err(e)) => ERR!("Error {} on getting instruction 0 of script {:?}", e, script),
+        None => ERR!("None instruction 0 of script {:?}", script),
+    }
+}
+
+/// Checks if a scriptSig is a script that spends a P2PK output.
+fn does_script_spend_p2pk(script: &Script) -> bool {
+    // P2PK scriptSig is just a single signature. The script should consist of a single push bytes
+    // instruction with the data as the signature.
+    extract_signature(script).is_ok() && script.get_instruction(1).is_none()
+}
+
+/// Verifies that the script that spends a P2PK is signed by the expected pubkey.
+fn verify_p2pk_input_pubkey(
+    script: &Script,
```

Comment:
why do we need to have `script` as a dedicated param? 
It is in the unsigned_tx already, right?

Reply:
that's the unlocking script (scriptSig). the `TransactionInputSigner` doesn't have such script (tx not signed yet).